app/controllers/passwords_controller.rb

```diff
@@ -0,0 +1,49 @@
+class PasswordsController < ApplicationController
+  def forgot_password
+  end
+
+  def request_token
+    @user = User.find_by(email_address: params[:email])
+    @token = SecureRandom.urlsafe_base64
+
+    if @user
+      @user.update_column(:token, @token)
+      UserMailer.reset_password(@user, @token).deliver
+      redirect_to confirm_password_reset_path
+    else
+      if params[:email].blank?
+        flash.now[:danger] = "Email cannot be blank!"
+      else
+        flash.now[:danger] = "User with this email does not exist!"
```

Comment:
Be careful with error messages that give out too much information. Many sites (including my own) say something like "If your email address is found in our system, you will receive instructions for resetting your password in a few minutes."